client/state/user-settings/actions.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import { isEmpty, keys } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import userUtils from 'lib/user/utils';
+import {
+	USER_SETTINGS_FETCH,
+	USER_SETTINGS_FETCH_SUCCESS,
+	USER_SETTINGS_FETCH_FAILURE,
+	USER_SETTINGS_SAVE_FAILURE,
+	USER_SETTINGS_SAVE_SUCCESS,
+	USER_SETTINGS_RECEIVE,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+
+const debug = debugFactory( 'calypso:user:settings' );
+
+/**
+ * Fetch user settings from WordPress.com API and store them in UserSettings instance
+ * @returns {Function} Action thunk
+ */
+export function fetchUserSettings() {
+	return ( dispatch, getState ) => {
+		if ( ! userUtils.isLoggedIn() || getState().userSettings.fetchingSettings ) {
+			return;
+		}
+
+		dispatch( { type: USER_SETTINGS_FETCH } );
+
+		debug( 'Fetching user settings' );
+
+		const fetchRequest = wpcom.me().settings().get();
+
+		fetchRequest.then( ( data ) => {
+			dispatch( receiveUserSettings( data ) );
+			dispatch( { type: USER_SETTINGS_FETCH_SUCCESS } );
+
+			debug( 'Settings successfully retrieved' );
+		} ).catch( ( error ) => {
+			dispatch( {
+				type: USER_SETTINGS_FETCH_FAILURE,
+				error
+			} );
+		} );
+
+		// Let the caller know whether the fetch was successful or not
+		return fetchRequest;
+	};
```

Comment:
I'll highly encourage a few things with this. Number one is to eliminate introducing a new Redux thunk action. Number two is to try writing this from inside the data layer. That _would_ require figuring out the actual API endpoint path and other settings specific to `wpcom.me().settings().get()` because the data layer prefers declarative HTTP request actions over direct stateful calling and promises..

In this process we can get rid of the abundance of extra meta actions ending in `_SUCCESS` and `_FAILURE`; also to change the nomenclature of `FETCH` to `REQUEST` and the semantics of `SETTINGS_RECEIVE` to something more meaningful in terms of Calypso intentions, like `SETTINGS_UPDATE`.